platforms/platforms.go

```diff
@@ -125,6 +126,12 @@ func (m *matcher) Match(platform specs.Platform) bool {
 		m.Variant == normalized.Variant
 }
 
+func (m *matcher) MatchCompat(platform specs.Platform) bool {
```

Comment:
I don't really want to introduce the concept of "compat" vs "exact" to the interface. That should be a function of the implementation of the interface.